### PR TITLE
allow decoration of workflow resources

### PR DESCRIPTION
**Pull request by danielnorberg**
The `WorkflowResourceDecorator` interface can be implemented to perform arbitrary modification of effective resource use by workflow instances at scheduler dequeue time.

This is effective regardless of trigger type.

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java [hunk 1]
```diff
@@ -7,9 +7,9 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
+ *
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java]: hahahaha

File: styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * -/-/-
+ */
+
+package com.spotify.styx;
+
+import com.spotify.styx.model.WorkflowConfiguration;
+import com.spotify.styx.model.WorkflowInstance;
+import java.util.Set;
+
+public interface WorkflowResourceDecorator {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java]: Do you think we should do some refactoring for the package? Some of the classes can be moved to `workflow`, e.g. this one, WorkflowCache, InMemWorkflowCache.

Maybe a separated PR.

**Comment by honnix** [comment]
minor comments, otherwise LGTM.

File: styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java (hunk 1 above)
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java]: *tableflip*

File: styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java (hunk 2 above)
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java]: Or maybe a `scheduler` package?

Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java]: But yes, could be another PR.

**Comment by honnix** [comment]
👍
